src/tests/multihost/sssd/testlib/common/utils.py

```diff
@@ -1576,6 +1576,38 @@ def create_ad_unix_user_group(self, username, groupname,
         mod = usr_res.returncode == 0
         return user and group and mbr and mod
 
+    def create_ad_unix_user(self, username, mail=None,
```

Comment:
ADOperations create_ad_unix_user_group? It creates user with group.

Reply:
The very nature of the range_retrival requirement is different. These tests requires:  
1. Single user who is part of 200 groups
2. Single group who has 200 users
3. A group with 50 AD-users members.
Using create_ad_unix_user_group would create additional 200 AD-groups(This is both resource and time consuming). These additional objects would not be used in any of the test case.
Thus creating a fixture to create single user is helpful here.
It's good to have a fixture which creates AD-posix_user on demand. There are tests which requires only user and not the group.
Alternative to create_ad_unix_user  would be to implement a powershell script to create, modify ADuser, and ADgroups objects as per requirement.

Reply:
Ok.